atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1891,6 +1897,121 @@ public void transactionStillCommittedEvenIfCallbackThrows() {
         });
     }
 
+    @Test
+    public void cannotUseCompletedFutureAfterTransactionCommit() throws Exception {
+        ListenableFuture<Map<Cell, byte[]>> getFuture = txManager.runTaskThrowOnConflict(txn -> {
+            txn.put(TABLE, ImmutableMap.of(TEST_CELL, PtBytes.toBytes("jolyon")));
```

Comment:
this is the second time my name features in a test!

Reply:
... I found the first one!

```
        assertThatThrownBy(() -> schema.addTableDefinition(TEST_TABLE_NAME, tableDef))
                .as("Jolyon loves this stuff")
                .isInstanceOf(SafeIllegalArgumentException.class)
                .hasMessage("Caching can only be enabled with the SERIALIZABLE_CELL conflict handler.");
```